### DISCUSSION TITLE
Get List of all contributors and their contents

### DISCUSSION THREAD

**Discussion by krishnakr6** [Sunbird-coKreat/Community#52] [General Discussions]
We need a list of all contributors and their contents on VDN -- Contributor Name, Individual/Org and Content.
Kindly let me know how to get these details.

**Comment by Aparna-karri591** [comment]
Hi Krishna,

Please go through this thread https://github.com/Sunbird-coKreat/Community/discussions/45, It would be helpful to you.

CC: @vinukumar-vs @mmanimegalai

**Reply by kameshbhr** [reply]
@Aparna-karri591 Thanks for the quick reply. Let me further clarify on the need.
We need contributor-wise number of contents contributed by each contributor which are published (that got approved by a sourcing org). Can you please let us know how to get this data? V_Org, V_User tables only have details of the contributing org/individual - not the number of contents created by them right?

**Comment by mmanimegalai** [comment]
@pallakartheekreddy Will you be able to provide the contributor and content list?

**Reply by pallakartheekreddy** [reply]
@mmanimegalai Currently, We don't have any API to get both Contributor Name, Individual/Org and Content details in one go. We need to use different API to get this. Please find details below
1. To get Contributor Name, Individual/Org we will use `Contribution Registry` API's. For details, please check [here](https://cokreat.sunbird.org/learn/product-and-developer-guide/asset-sourcing/contribution-registry)
2. To get the list of contents, what contributed by each user use content search API. This will give you all the contents means approved and non-approved contents. For details about search API click [here](https://documenter.getpostman.com/view/25463377/2s8ZDa3MP7)
3. To get the list of contents which are published (that got approved by a sourcing org). We need to use search API of consumption repo because we are storing contents details which are published from sourcing org in different places. 
For project type `Get content targeted to any collections` we are storing published contents details at each collection level
For project type `Get individual content not targeted to any collection` we are storing published contents details at each project level
So because of this, We need to use search API of consumption repo with below request to get the list of contents which are published from sourcing.

```{"request":{"filters":{"objectType":["content"],"status":["Live"],"mimeType":{"!=":"application/vnd.ekstep.content-collection"},"contentType":{"!=":"Asset"}},"exists":["programId","originData"],"fields":["name","status","primaryCategory","channel","originData"],"limit":1}}```

**Comment by krishnakr6** [comment]
@pallakartheekreddy content search and search API of consumption repo are different APIs? If so , could you share the relevant API endpoints

**Reply by pallakartheekreddy** [reply]
@krishnakr6 both are same only.

**Reply by krishnakr6** [reply]
@pallakartheekreddy thank you.

**Reply by krishnakr6** [reply]
@pallakartheekreddy could you please share  one sample request for contributor registry API, **/search** endpoint.

**Request**
```
curl --location 'https://<url>/api/search' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer <token>' \
--data '{
  "id": "open-saber.registry.search",
  "ver": "1.0",
  "ets": "11234",
  "params": {
    "did": "",
    "key": "",
    "msgid": ""
  },
  "request": {
    "opensaberapientityType": [
      "Teacher"
    ],
    "opensaberapifilters": {
      "gender": {
        "eq": "GenderTypeCode-MALE"
      }
    }
  }
}'
```
**Response**
```
{
    "message": "no route and no API found with those values"
}
```

**Reply by vaivk369** [reply]
@krishnakr6 Please refer this:
`curl --location '<HOST>/content/reg/search' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer <token>' \
--data ' {
    "id":"open-saber.registry.search",
    "ver":"1.0",
    "ets":"11234",
    "params":
              {
                 "did":"",
                 "key":"",
                 "msgid":""
               },
     "request": {
            "entityType":["Org"],
             "filters": {
                    "orgId": {
                         "eq":"01346442741037465621"
                     }
             }
        }
   }' \
  --compressed`

**Reply by krishnakr6** [reply]
@vaivk369 ok thanks, let me check.

**Reply by krishnakr6** [reply]
@vaivk369 this **endpoint : /content/reg/search** is not available in  https://github.com/project-sunbird/sunbird-devops/blob/release-5.1.0/ansible/roles/kong-api/defaults/main.yml
@pallakartheekreddy action/composite/v3/search is also available.
 could you provide the working endpoints, with a sample request.

**Reply by krishnakr6** [reply]
@pallakartheekreddy  any update?

**Comment by mmanimegalai** [comment]
@vaivk369 Can you please reply to the above request?
cc @pallakartheekreddy @vinukumar-vs
